front/app/modules/commercial/project_folders/citizen/components/ProjectFolderGoBackButton/index.test.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { render, screen, fireEvent } from 'utils/testUtils/rtl';
+
+import GoBackButton from './';
+
+const projectFolderData = {
+  id: 'a8735d6f-cdd4-4242-8531-b089b5e3954a',
+  type: 'folder',
+  attributes: {
+    title_multiloc: {
+      en: 'TestFolder',
+      'fr-BE': 'Le folder du test',
+      'nl-BE': 'TestMap',
+    },
+    slug: 'testfolder',
+  },
+};
+
+let historyState: string;
+const history = {
+  push(location: string) {
+    historyState = location;
+  },
+};
+
+jest.mock('utils/cl-router/history', () => history);
+jest.mock('../../../hooks/useProjectFolder', () =>
+  jest.fn(() => projectFolderData)
+);
+jest.mock('hooks/useLocale', () => jest.fn(() => 'en'));
+jest.mock('hooks/useLocalize', () => jest.fn(() => (multiloc) => multiloc.en));
+
+describe('GoBackButton', () => {
+  it('should render', () => {
+    render(<GoBackButton projectFolderId={projectFolderData.id} />);
+    expect(screen.getByRole('button')).toBeInTheDocument();
+    expect(screen.getByText('TestFolder')).toBeInTheDocument();
+  });
+  it('pushes parent folder path to history when clicked', () => {
```

Comment:
The way that we have tested history changes in other tests is the following:
```
import clHistory from 'utils/cl-router/history';

jest.mock('utils/cl-router/history'); // use our already existing mock

// Test push gets called correctly
  it('pushes parent folder path to history when clicked', () => {
    render(<GoBackButton projectFolderId={projectFolderData.id} />);

    const button = screen.getByRole('button');
    fireEvent.click(button);

    expect(clHistory.push).toHaveBeenCalledWith('/folders/testfolder');
  });
```

What you are proposing works as well, it is just a bit more verbose to write and doesn't take advantage of what we already have in place as mocks. :)